<!DOCTYPE html>
<html lang="en">
  <head>
    <title>
  Software · LRSlab@CityU
</title>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="LRS-Lab">
<meta name="description" content="Hammerhead - [Python] - Hammerhead is a pipeline for the detection of bacterial DNA modifications and fix modification induced assembly error from Nanopore R10.4.1 reads. Published in Genome Research, 2024
nanoCEM -[Python] - The nanopore current events magnifier (nanoCEM) is a python command line to facilitate the analysis of DNA/RNA modification sites by visualizing statistical features of current events. Published in NARGAB, 2024
Giraffe - [Python] - Giraffe is used for long read QC, can be used to calculate and visualize basic statistics for nanopore/pacbio long reads, including observed/estimated read quality; gc bias for reads; homopolymer accuracy and others.">
<meta name="keywords" content="Bioinformatics,Long read sequencing, DNA/RNA modification, Caenorhabditis elegans, Genetics">

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Software"/>
<meta name="twitter:description" content="Hammerhead - [Python] - Hammerhead is a pipeline for the detection of bacterial DNA modifications and fix modification induced assembly error from Nanopore R10.4.1 reads. Published in Genome Research, 2024
nanoCEM -[Python] - The nanopore current events magnifier (nanoCEM) is a python command line to facilitate the analysis of DNA/RNA modification sites by visualizing statistical features of current events. Published in NARGAB, 2024
Giraffe - [Python] - Giraffe is used for long read QC, can be used to calculate and visualize basic statistics for nanopore/pacbio long reads, including observed/estimated read quality; gc bias for reads; homopolymer accuracy and others."/>

<meta property="og:title" content="Software" />
<meta property="og:description" content="Hammerhead - [Python] - Hammerhead is a pipeline for the detection of bacterial DNA modifications and fix modification induced assembly error from Nanopore R10.4.1 reads. Published in Genome Research, 2024
nanoCEM -[Python] - The nanopore current events magnifier (nanoCEM) is a python command line to facilitate the analysis of DNA/RNA modification sites by visualizing statistical features of current events. Published in NARGAB, 2024
Giraffe - [Python] - Giraffe is used for long read QC, can be used to calculate and visualize basic statistics for nanopore/pacbio long reads, including observed/estimated read quality; gc bias for reads; homopolymer accuracy and others." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://lrslab.github.io/software/" /><meta property="article:section" content="" />






<link rel="canonical" href="http://lrslab.github.io/software/">


<link rel="preload" href="/fonts/forkawesome-webfont.woff2?v=1.2.0" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.7f9d180e3b3bebba9ba80d55eed1255c35e00764872854736d6ad7db38884ffc.css" integrity="sha256-f50YDjs767qbqA1V7tElXDXgB2SHKFRzbWrX2ziIT/w=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">




<meta name="generator" content="Hugo 0.108.0">





  </head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    <a class="navigation-title" href="/">
      LRSlab@CityU
    </a>
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link" href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/software/">Software</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://lrslab.github.io/software/">
          Software
        </a>
      </h1>
    </header>

    <ul>
<li>
<p><a href="https://github.com/lrslab/Hammerhead">Hammerhead</a> - [Python] - Hammerhead is a pipeline for the detection of bacterial DNA modifications and fix modification induced assembly error from Nanopore R10.4.1 reads.  <a href="https://genome.cshlp.org/content/early/2024/10/02/gr.279012.124.abstract">Published in <em><strong>Genome Research, 2024</strong></em></a></p>
<ul>
<li>
<img src="https://raw.githubusercontent.com/lrslab/Hammerhead/main/figure_demo/Demo_1.png" width='300'> 
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/lrslab/nanoCEM">nanoCEM</a> -[Python] - The nanopore current events magnifier (nanoCEM) is a python command line to facilitate the analysis of DNA/RNA modification sites by visualizing statistical features of current events.  <a href="https://doi.org/10.1093/nargab/lqae052">Published in <em><strong>NARGAB, 2024</strong></em></a></p>
<ul>
<li>
<img src="https://raw.githubusercontent.com/lrslab/nanoCEM/main/docs/Workflow.png" width='300'> 
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/lrslab/Giraffe_View">Giraffe</a> - [Python] - Giraffe is used for long read QC, can be used to calculate and visualize basic statistics for nanopore/pacbio long reads, including observed/estimated read quality; gc bias for reads; homopolymer accuracy and others.<a href="https://www.sciencedirect.com/science/article/pii/S2001037024002629">Published in <em><strong>CSBJ 2024</strong></em></a>
Some of these features has been used in our benchmarking papers on <a href="https://www.sciencedirect.com/science/article/pii/S2001037023001368">single cell DNA sequencing</a> and <a href="https://www.nature.com/articles/s41597-023-02431-5">DNA methylation profiling</a>.</p>
<ul>
<li>
<img src="https://ars.els-cdn.com/content/image/1-s2.0-S2001037024002629-gr1.jpg" width='300'>
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/lrslab/anasfv">ANASFV</a> - [Python] - ANASFV is a pipeline used for the analysis of african swine fever virus. [Manuscript preprint] (<a href="https://www.biorxiv.org/content/10.1101/2024.07.08.602471v1">https://www.biorxiv.org/content/10.1101/2024.07.08.602471v1</a>)</p>
</li>
<li>
<p><a href="https://github.com/runsheng/trackcluster">trackcluster</a> -[Python] - trackcluster is an isoform calling and quantification pipeline for long RNA/cDNA reads. <a href="https://genome.cshlp.org/content/30/2/287.short">Published in <em><strong>Genome Research, 2020</strong></em></a></p>
<ul>
<li>
<img src="https://genome.cshlp.org/content/30/2/287/F2.large.jpg"  width="300">
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/runsheng/mitovar">mitovar</a> -[Python] - mitovar is a pipeline used for mtDNA genome assembly from NGS reads.  <a href="https://academic.oup.com/dnaresearch/article/25/6/577/5066955">Published in <em><strong>DNA Research, 2018</strong></em></a></p>
<ul>
<li>
<img src="/images/mitovar_dsy026f1.jpeg" width='300'> 
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/runsheng/primerdiffer">primerdiffer</a> -[Python] - primerdiffer is a pipeline used to design primers in a large batch. <a href="https://academic.oup.com/bioinformatics/article/39/4/btad188/7126407">Published in <em><strong>Bioinformatics, 2023</strong></em></a></p>
<ul>
<li><a href="https://github.com/runsheng/primervcf">primervcf</a> - [Python] - primervcf is a sub-module for primerdiffer, to design haplotyping primers using vcf file.</li>
<li>
<img src="/images/primerdiffer_btad188f1.jpeg" width='300'>
</li>
</ul>
</li>
</ul>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    ©
    
      2020 -
    
    2025
     LRS-Lab 
    ·
    
    Long Read Sequencing Lab
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.27afce394fb6284f521b3fbc9f6a8326342333c3092267f3944d770489876fed.js" integrity="sha256-J6/OOU&#43;2KE9SGz&#43;8n2qDJjQjM8MJImfzlE13BImHb&#43;0="></script>
  

  

  

  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  
</body>

</html>
